fcrepo-exts/fcrepo-migration-validator
# Compare DateTimes as Instants intead of Strings

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3715

# What does this Pull Request do?

This updates the comparison of DateTimes in the Object metadata to be based on Instants rather than Strings. This is done in order to account for potential differences in how the DateTime was formatted.

# What's new?

* Compare DateTimes using Instants
* Update and remigrated test data for ObjectValidationIT to have DateTimes at `000Z`

# How should this be tested?

Migrate a data set with DateTimes which are formatted similar to `2019-07-08T18:49:12.000Z` in Fedora 3 and look like `2019-07-08T18:49:12Z` in Fedora 6 then run the validator and ensure that there are no failures.

# Additional Notes

I removed some of the typing from the PropertyResolver interface because we had previously been doing everything on Strings, and trying to move back to a generic type ended up making things messy without making more changes. In the interest of not having a bunch of churn in order to accomplish this, I decided to treat everything as a String in the top level PropertyResolver and create a DateTimeResolver which could convert the Strings back to Instants when doing the comparison. 

# Interested parties

@dbernstein
